### PR TITLE
Add Junit5 extension for Influx Server instance fields

### PR DESCRIPTION
**Pull request by trav**
Add Junit5 extension for Influx Server instance fields

- Also fixed build which was not actually running the tests for the embedded-influx-junit5 module

### CODE REVIEW

**Comment by trav** [comment]
@benjamin-bader -- this extension deviates from how you were doing the Builder in the JUnit4 rule -- you might have an opinion about that.  I'm also up for changing the name of the extension.

File: build.gradle [hunk 1]
```diff
@@ -62,7 +62,7 @@ subprojects {
                 ],
                 junit4: [ "junit:junit:$junit4Version" ],
                 junit5: [
-                        "org.junit:junit-bom:$junit5Version"
+                        "org.junit.jupiter:junit-jupiter-engine:$junit5Version"
```
Comment (benjamin-bader) [inline build.gradle:65]: Just curious, why is this needed?  Wouldn't the engine come as part of the bom dependency?

File: embedded-influx-junit5/src/main/kotlin/com/bendb/influx/InfluxServerInstance.kt [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * Example usage:
+ *
+ * ```
+ * class ExampleTest {
+ *     @JvmField
+ *     @RegisterExtension
+ *     val server = InfluxServerInstance(InfluxServerBuilder().timeout(Duration.ofSeconds(10)))
+ *
+ *     @Test fun pingThatServer() {
+ *         InfluxDBFactory.connect(server.url).use { client ->
```
Comment (benjamin-bader) [inline embedded-influx-junit5/src/main/kotlin/com/bendb/influx/InfluxServerInstance.kt:40]: nit: missing a closing brace and indentation for the `use` block in the comment.

File: embedded-influx-junit5/src/main/kotlin/com/bendb/influx/InfluxServerInstance.kt [hunk 3]
```diff
@@ -0,0 +1,47 @@
+package com.bendb.influx
```
Comment (benjamin-bader) [inline embedded-influx-junit5/src/main/kotlin/com/bendb/influx/InfluxServerInstance.kt:17]: nit: missing license header

File: embedded-influx-junit5/build.gradle [hunk 4]
```diff
@@ -23,4 +23,10 @@ dependencies {
     testImplementation libs.influxClient
     testImplementation libs.junit5
     testImplementation libs.kotlinTest
-}
\ No newline at end of file
+    
+}
+
+test {
+    useJUnitPlatform {
```
Comment (benjamin-bader) [inline embedded-influx-junit5/build.gradle:30]: Also unclear on what this does; don't tests run correctly already?

Comment (trav) [reply embedded-influx-junit5/build.gradle:30]: The JUnit 5 tests won't run without it.  If you add a `TODO` to a test in the JUnit5 module, then run `./gradlew clean test`, it won't blow up the build.

File: build.gradle (hunk 1 above)
Comment (trav) [reply build.gradle:65]: Without this change, you see this when running the tests in the junit5 directory:  
```
Caused by: org.junit.platform.commons.util.PreconditionViolationException: Cannot create Launcher without at least one TestEngine; consider adding an engine implementation JAR to the classpath
```

Comment (trav) [reply build.gradle:65]: If you run `./gradlew dependencies` in the junit5 directory, you see the bom getting imported, but none of its dependencies...

File: embedded-influx-junit5/build.gradle (hunk 4 above)
Comment (benjamin-bader) [reply embedded-influx-junit5/build.gradle:30]: Huh, TIL.

**Review by benjamin-bader — APPROVED** [review]

**Comment by benjamin-bader** [comment]
Thanks!  I've just pushed version 0.2.0 to Maven Central, and it ought to be available within the next ~15 minutes or so.
